chunks.py

```diff
@@ -0,0 +1,130 @@
+import asyncio
+import io
+import math
+import os
+from concurrent.futures import ThreadPoolExecutor
+
+import config
+from progresslogger import Progress
+
+
+def calculate_number_of_chunks(file_size, chunk_size):
+    return math.ceil(int(file_size) / chunk_size)
+
+
+class Chunks:
+    def __init__(self, file_downloader, file_name, file_size, *chunks: 'Chunk'):
+        def schedule(chunk):
+            return loop.run_in_executor(executor, chunk.download, file_downloader)
+
+        async def write_files():
+            async def await_chunks():
+                for chunk in chunks:
+                    await chunk
+
+            def write_final_file_from_partial_files():
+                with open(file_name, "wb") as final_file:
+                    for chunk in chunks:
+                        with open(chunk.file_name, "rb") as partial_file:
+                            final_file.write(partial_file.read())
+
+            def remove_partial_files():
```

Comment:
I tested interrupting a download and it left part 31 here somehow  
It threw a lot of errors, and this one was at the end   
During handling of the above exception, another exception occurred:

```
Traceback (most recent call last):
  File "/home/jrsc/Documents/gdrive-manager/chunks.py", line 39, in write_files
    remove_partial_files()
  File "/home/jrsc/Documents/gdrive-manager/chunks.py", line 33, in remove_partial_files
    os.remove(chunk.file_name)
FileNotFoundError: [Errno 2] No such file or directory: 'fastboot_sofia_vzw_userdebug_11_RPM31.107_eng.jcamelo.210415.165840_intcfg_test-keys_verizon_US.tar.gz.29'
```

Reply:
Were all other parts removed?

Reply:
Yes, only part 31 left

Reply:
Fixed retro-compatibility with 3.6. Please retest.

Reply:
Fixed, please recheck and resolve if all is ok.